### PR TITLE
Search: add query param to prepopulate search text

### PR DESCRIPTION
**Pull request by MisterSquishy**
**What this PR does / why we need it**:
adds a query parameter to prepopulate  dashboard search text when using the `seach=open` param. this enables folks to make a [custom search engine](https://www.techrepublic.com/article/pro-tip-add-custom-search-engines-in-chrome-for-more-efficient-searching/) to make searching for dashboards super easy

no filter:
image

just folder (side note -- folders with spaces dont seem to behave super nicely here; i.e. you cant actually search within them. seems like that behavior exists on master so calling that out of scope for this PR):
image

just query:
image

folder + query:
image

closing cleans up the url:
2020-06-12 at 11 56

### CODE REVIEW

File: e2e/suite1/specs/query-editor.spec.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ e2e.scenario({
 
     cy.contains(queryText.slice(0, -1)).should('be.visible');
 
-    e2e.components.QueryField.container().type('{ctrl}z');
+    e2e.components.QueryField.container().type('{cmd}z');
```
Comment (MisterSquishy) [inline e2e/suite1/specs/query-editor.spec.ts:32]: as noted in the description, this is unrelated. at the ready to revert if it breaks on CI or something

**Comment by Clarity-89** [comment]
There's a similar PR already that also saves filters to the URL params: https://github.com/grafana/grafana/pull/25541. Good idea with the tests though, I'll need to add them to my PR as well.

**Comment by dprokop** [comment]
@MisterSquishy thanks for the contribution! @Clarity-89 do you want this to be merged, or you will continue in your PR? If so, let's close this one.

**Comment by Clarity-89** [comment]
@dprokop let's close this one.

**Pull request closed without merging**